static/app/components/searchSyntax/parser.tsx

```diff
@@ -878,6 +890,82 @@ export class TokenConverter {
   };
 }
 
+function parseDate(input: string): Date {
+  const date = new Date(input);
+
+  if (isNaN(date.getTime())) {
+    throw new Error('Invalid date');
+  }
+
+  return date;
+}
+
+function parseRelativeDate(
+  input: string,
+  {sign, unit}: {sign: '-' | '+'; unit: string}
+): Date {
+  let date = new Date(input).getTime();
+
+  if (isNaN(date)) {
+    throw new Error('Invalid date');
+  }
+
+  let offset: number | undefined;
+  switch (unit) {
+    case 'm':
+      offset = 1000 * 60;
+      break;
+    case 'h':
+      offset = 1000 * 60 * 60;
+      break;
+    case 'd':
+      offset = 1000 * 60 * 60 * 24;
+      break;
+    case 'w':
+      offset = 1000 * 60 * 60 * 24 * 7;
+      break;
+    default:
+      throw new Error('Invalid unit');
+  }
+
+  if (offset === undefined) {
+    throw new Error('Unreachable');
+  }
+
+  date = sign === '-' ? date - offset : date + offset;
+  return new Date(date);
+}
+
+// The parser supports floats and ints, parseFloat handles both.
+function numeric(input: string) {
+  const number = parseFloat(input);
+  if (isNaN(number)) {
+    throw new Error('Invalid number');
+  }
+  return number;
+}
+function parseDuration(input: string, _unit: string): number {
+  return numeric(input);
+}
```

Comment:
Why isn't the unit used?

Reply:
We would need to make a decision of what unit to what base unit we want to normalize to. This just parses the value so that whoever is consuming this can make the decision of what base to normalize this to

Reply:
we should just parse down to milliseconds no?

Reply:
Yes, but I cant say if thats what everyone would want, so I'd rather not make that decision at all and keep it open. We would need to make the equivalent decisions for sizes and other units and I'm not sure yet if it's something we can make a decision about

Reply:
This feels more right to me to normalize it though. I don't want to have to do the work to parse the units, as a user of the token results I would expect the parsed value to be normalized down.

IMO we should be doing the same for size units, it should be parsed down to bytes

Reply:
Yeah, I agree, it's just that I dont know if I know enough to make that call 😂